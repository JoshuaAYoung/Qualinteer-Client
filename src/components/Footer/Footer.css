.footer {
  width: 100%;
  text-align: center;
  margin: 0;
  height: auto;
  position: absolute;
  bottom: 0;
}

.footerInside {
  background: #F9EDBD;
  height: 100%;
}

.socialImage {
  width: 4.5rem;
}

.socialContainer {
  display: flex;
  justify-content: space-evenly;
  padding-top: 1.5rem;
}

.footerWaveContainer {
  overflow: hidden;
  height: 6.8rem;
  margin-bottom: -1rem;
}

.footerWave {
  height: 6.8rem;
  object-position: -10rem;
}

.copyright {
  margin-top: 1em;
  padding-bottom: 1em;
  font-size: 1.4rem;
}
