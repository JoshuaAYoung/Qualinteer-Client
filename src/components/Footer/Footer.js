import React from 'react';

export default function Footer() {
  function generateCopyright() {
    let d = new Date();
    let year = d.getFullYear();
    return year;
  }

  return (
    <footer className='copyright'>
      <p>
        Copyright &copy; {generateCopyright()}
        <a href='https://joshyoung.net' target='_blank' rel='noopener noreferrer' className='boldLink'>Josh Young</a>
      </p>
    </footer>
  );
}
