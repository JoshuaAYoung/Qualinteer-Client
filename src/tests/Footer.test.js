import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Footer from '../components/Footer/Footer'

describe('Footer Component', () => {

  it('Footer Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<Footer />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Footer Snapshot Test', () => {
    const wrapper = shallow(
      <Footer />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})