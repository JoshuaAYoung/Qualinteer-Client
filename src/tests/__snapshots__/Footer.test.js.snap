// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Footer Snapshot Test 1`] = `
<footer
  className="footer"
>
  <div
    className="footerWaveContainer"
  >
    <img
      alt="wavy border"
      className="footerWave"
      src="/assets/top-waves.svg"
    />
  </div>
  <div
    className="footerInside"
  >
    <div
      className="socialContainer"
    >
      <a
        className="mailToLink"
        href="mailto:joshua@young.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="email logo"
          className="socialImage"
          src="/assets/email-logo.svg"
        />
      </a>
      <a
        className="githubLink"
        href="https://github.com/JoshuaAYoung"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="github logo"
          className="socialImage"
          src="/assets/github-logo.svg"
        />
      </a>
      <a
        className="linkedinLink"
        href="https://www.linkedin.com/in/joshuaalexanderyoung"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="linkedin logo"
          className="socialImage"
          src="/assets/linkedin-logo.svg"
        />
      </a>
    </div>
    <p
      className="copyright"
    >
      Copyright Â© 
      2020
      <br />
      <a
        className="portLink"
        href="https://joshyoung.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        Josh Young
      </a>
    </p>
  </div>
</footer>
`;
